### PR TITLE
Aryeh/110674044 crash fix

### PR DESCRIPTION
**Pull request by katzio**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -110,7 +113,8 @@ dependencies {
     compileOnly 'org.glassfish:javax.annotation:10.0-b28'
     kapt 'com.google.dagger:dagger-android-processor:2.15'
     kapt 'com.google.dagger:dagger-compiler:2.15'
-    implementation 'com.google.dagger:dagger-android-support:2.15' // if you use the support libraries
+    implementation 'com.google.dagger:dagger-android-support:2.15'
+    // if you use the support libraries
```
Comment (berryve) [inline app/build.gradle]: can we remove this comment ?

File: app/src/main/java/org/kinecosystem/kinit/view/adapter/OfferListAdapter.kt [hunk 2]
```diff
@@ -14,7 +14,8 @@ import org.kinecosystem.kinit.model.spend.isP2p
 import org.kinecosystem.kinit.util.ImageUtils
 import org.kinecosystem.kinit.viewmodel.spend.SpendViewModel
 
-class OfferListAdapter(private val context: Context, private val model: SpendViewModel) : RecyclerView.Adapter<OfferListAdapter.ViewHolder>() {
+class OfferListAdapter(private val context: Context, private val model: SpendViewModel)
+    : RecyclerView.Adapter<OfferListAdapter.ViewHolder>() {
 
 
     private var offerList: List<Offer> = model.offers()
```
Comment (katzio) [inline app/src/main/java/org/kinecosystem/kinit/view/adapter/OfferListAdapter.kt]: Shay:can u change the name not to contian list

File: app/src/main/java/org/kinecosystem/kinit/view/spend/PurchaseOfferActivity.java [hunk 3]
```diff
@@ -5,26 +5,28 @@
...
 
 public class PurchaseOfferActivity extends BaseActivity {
 
     private static final String INDEX_PARAM = "SpendActivity_index_param";
 
-    public static Intent getIntent(Context context, int index) {
+    public static Intent getIntent(Context context, Offer offer) {
         Intent intent = new Intent(context, PurchaseOfferActivity.class);
-        intent.putExtra(INDEX_PARAM, index);
+        intent.putExtra(INDEX_PARAM, offer);
```
Comment (katzio) [inline app/src/main/java/org/kinecosystem/kinit/view/spend/PurchaseOfferActivity.java]: offer param

File: app/src/main/java/org/kinecosystem/kinit/view/spend/PurchaseOfferFragment.java [hunk 4]
```diff
@@ -26,44 +27,38 @@
 
 public class PurchaseOfferFragment extends BaseFragment implements PurchaseOfferActions {
 
-    public static final String ARG_OFFER_INDEX = "SpendOfferFragment_ARG_QUESTION_INDEX";
-    public static final int INVALID_OFFER_INDEX = -1;
+    public static final String ARG_OFFER = "SpendOfferFragment_ARG_QUESTION_INDEX";
```
Comment (katzio) [inline app/src/main/java/org/kinecosystem/kinit/view/spend/PurchaseOfferFragment.java]: value not correct

File: app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/PurchaseOfferViewModel.kt [hunk 5]
```diff
@@ -48,7 +47,6 @@ class PurchaseOfferViewModel(private val navigator: Navigator, offerIndex: Int)
 
     init {
         KinitApplication.coreComponent.inject(this)
```
Comment (katzio) [inline app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/PurchaseOfferViewModel.kt]: Shay remove offerrepo

**Pull request merged**